dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/workflow/execution/WorkflowExecutionWorkerTest.java

```diff
@@ -0,0 +1,288 @@
+import edu.harvard.iq.dataverse.persistence.user.AuthenticatedUser;
+import edu.harvard.iq.dataverse.persistence.workflow.Workflow;
+import edu.harvard.iq.dataverse.persistence.workflow.WorkflowExecution;
+import edu.harvard.iq.dataverse.persistence.workflow.WorkflowExecutionRepository;
+import edu.harvard.iq.dataverse.persistence.workflow.WorkflowExecutionStep;
+import edu.harvard.iq.dataverse.persistence.workflow.WorkflowRepository;
+import edu.harvard.iq.dataverse.settings.SettingsServiceBean;
+import edu.harvard.iq.dataverse.workflow.WorkflowStepRegistry;
+import edu.harvard.iq.dataverse.workflow.WorkflowStepSPI;
+import edu.harvard.iq.dataverse.workflow.listener.WorkflowExecutionListener;
+import edu.harvard.iq.dataverse.workflow.step.Failure;
+import edu.harvard.iq.dataverse.workflow.step.WorkflowStep;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import javax.enterprise.inject.Instance;
+import javax.naming.NamingException;
+import java.time.Clock;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static edu.harvard.iq.dataverse.persistence.dataset.DatasetMother.givenDataset;
+import static edu.harvard.iq.dataverse.persistence.workflow.WorkflowMother.givenWorkflow;
+import static edu.harvard.iq.dataverse.persistence.workflow.WorkflowMother.givenWorkflowStep;
+import static edu.harvard.iq.dataverse.workflow.execution.WorkflowContextMother.givenWorkflowExecutionContext;
+import static java.time.ZoneOffset.UTC;
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.singletonMap;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+
+class WorkflowExecutionWorkerTest extends WorkflowJMSTestBase implements WorkflowStepSPI {
+
+    static final String TEST_PROVIDER_ID = "test";
+    static final String SUCCESS_STEP_ID = "successful";
+    static final String PAUSING_STEP_ID = "pausing";
+    static final String FAILING_STEP_ID = "failing";
+
+    Clock clock = Clock.fixed(Instant.parse("2020-06-01T09:10:20.00Z"), UTC);
+
+    SettingsServiceBean settings = new TestSettingsServiceBean();
+    StubJpaPersistence persistence = new StubJpaPersistence();
+    DatasetRepository datasets = persistence.stub(DatasetRepository.class);
+    WorkflowRepository workflows = persistence.stub(WorkflowRepository.class);
+    WorkflowExecutionRepository executions = persistence.stub(WorkflowExecutionRepository.class);
+    WorkflowStepRegistry steps = new WorkflowStepRegistry() {{ init(); }};
+    RoleAssigneeServiceBean roleAssignees = new MockRoleAssigneeServiceBean() {{ add(new MockAuthenticatedUser()); }};
+    AuthenticationServiceBean authentication = new MockAuthenticationServiceBean(clock);
```

Comment:
IMO we should name repositories with `repository` postfix and `ServiceBean` maybe with `Service` postfix.
When we read test methods then it is really easy to get confused with model classes.
For example list of executions and executionRepository

Reply:
I admit the name conflict with the local variable in test methods is bit unfortunate and I fixed it. My reasoning for not suffixing the variable names with `Repository` or `Service` is that withing the context those variables are used it doesn't really matter. That matters is that they are an interface over a collection of elements, like workflows or datasets or any other database table representing a persistent collection. In the context they are used I usually don't mind if there is some extra logic around that collection, like in a service, or if it's a mostly thin access object like repository. I only care about what the result of a query method call on that collection is or that I wan to pass something into that collection to mutate it. Those are ultimately the two types of operation there are. I'd give more descriptive names to local variables representing subsets of those collections that would describe what kind of subset they represent, rather than enforce an artificial suffix on the name of the collection itself.

Reply:
I must say that I'm not a super fan of this approach. It introduces a lot of confusion when you working with lists in methods. And we could argue that everything is an "interface over collection". But considering that the whole workflow mechanism is written in such style. I would approve this.